pyro/contrib/funsor/handlers/plate_messenger.py

```diff
@@ -50,7 +47,7 @@ def __enter__(self):
         name_to_dim = OrderedDict([(self.name, DimRequest(self.dim, self.dim_type))])
         indices = to_data(self._indices, name_to_dim=name_to_dim)
         # extract the dimension allocated by to_data to match plate's current behavior
-        self.dim, self.indices = -indices.dim(), indices.squeeze()
+        self.dim, self.indices = -indices.dim(), indices.reshape(-1)
```

Comment:
Changed to `.reshape(-1)` so that `self.indices` would have a vector shape even for `self.size=1`.  This makes it compatible with the behavior of the pyro backend.